### PR TITLE
Clarifies some silicon interactions via screentips, also converts some stuff to balloon alerts

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

APCs, fire alarms, and holopads now communicate a few more hotkeys for interaction with silicons via examine. Ex. APCs now tell silicons they can disable lighting with shift-click. 
## Why It's Good For The Game

1. Balloon alerts are cool.
2. UX is also cool.
## Changelog
:cl:
qol: APCs, fire alarms, and holopads now communicate some more of their silicon interactions via screentips.
qol: APCs, fire alarms, holopads and turret control panels now use balloon alerts for more of their notifications.
/:cl:

### CODE REVIEW

File: code/_onclick/ai.dm [hunk 1]
```diff
@@ -197,9 +197,12 @@
 	if(!is_operational || failure_timer)
 		return
 
-	add_hiddenprint(user)
 	environ = environ ? APC_CHANNEL_OFF : APC_CHANNEL_ON
-	user.log_message("turned [environ ? "on" : "off"] the [src] environment settings", LOG_GAME)
+	if (user)
+		add_hiddenprint(user)
+		var/on_or_off = environ ? "on" : "off"
+		balloon_alert(user, "environment power toggled [on_or_off]")
```
Comment (Mothblocks) [inline code/_onclick/ai.dm:204]: Nit: I usually like `enabled/disabled` in balloon alerts

File: code/_onclick/ai.dm [hunk 2]
```diff
@@ -211,10 +214,13 @@
...
 		return
 
-	add_hiddenprint(user)
 	lighting = lighting ? APC_CHANNEL_OFF : APC_CHANNEL_ON
-	user.log_message("turned [lighting ? "on" : "off"] the [src] lighting settings", LOG_GAME)
-	update_appearance()
+	if (user)
+		var/on_or_off = lighting ? "on" : "off"
+		add_hiddenprint(user)
+		balloon_alert(user, "lighting power toggled [on_or_off]")
```
Comment (Mothblocks) [inline code/_onclick/ai.dm:221]: Same nit, it makes these nicer to read

File: code/game/machinery/firealarm.dm [hunk 3]
```diff
@@ -460,6 +460,9 @@
 		. += "The local area thermal detection light is [my_area.fire_detect ? "lit" : "unlit"]."
 		. += "<b>Left-Click</b> to activate all firelocks in this area."
 
+	if (issilicon(user))
+		. += span_notice("Ctrl-Click the fire alarm to toggle thermal sensors, which controls auto-activation.")
```
Comment (Mothblocks) [inline code/game/machinery/firealarm.dm:464]: This should be a contextual screentip. We should add those to fire alarms if we don't already

File: code/game/machinery/firealarm.dm [hunk 4]
```diff
@@ -477,8 +480,10 @@
 	my_area.fire_detect = !my_area.fire_detect
 	for(var/obj/machinery/firealarm/fire_panel in my_area.firealarms)
 		fire_panel.update_icon()
-	to_chat(user, span_notice("You [ my_area.fire_detect ? "enable" : "disable" ] the local firelock thermal sensors!"))
-	user.log_message("[ my_area.fire_detect ? "enabled" : "disabled" ] firelock sensors using [src].", LOG_GAME)
+	if (user)
+		balloon_alert(user, "thermal sensors [my_area.fire_detect ? "enabled" : "disabled"]")
+		user.log_message("[ my_area.fire_detect ? "enabled" : "disabled" ] firelock sensors using [src].", LOG_GAME)
+	//to_chat(user, span_notice("You [ my_area.fire_detect ? "enable" : "disable" ] the local firelock thermal sensors!"))
```
Comment (Mothblocks) [inline code/game/machinery/firealarm.dm:486]: Commented code?

File: code/modules/power/apc/apc_power_proc.dm [hunk 5]
```diff
@@ -42,8 +42,11 @@
 	if(!is_operational || failure_timer)
 		return
 	operating = !operating
-	add_hiddenprint(user)
-	user.log_message("turned [operating ? "on" : "off"] the [src]", LOG_GAME)
+	if (user)
+		var/on_or_off = operating ? "on" : "off"
+		balloon_alert(user, "power toggled [on_or_off]")
```
Comment (Mothblocks) [inline code/modules/power/apc/apc_power_proc.dm:47]: Same nit

File: code/modules/power/apc/apc_main.dm [hunk 6]
```diff
@@ -288,6 +288,9 @@
 
 	if(issilicon(user))
 		. += span_notice("Ctrl-Click the APC to switch the breaker [ operating ? "off" : "on"].")
+		. += span_notice("Alt-Click the APC to toggle equipment power.")
+		. += span_notice("Shift-Click the APC to toggle lighting power.")
+		. += span_notice("Shift+Ctrl-Click the APC to toggle environment power.")
```
Comment (Mothblocks) [inline code/modules/power/apc/apc_main.dm:293]: These should be contextual screentips

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Small nits, looks good

File: code/game/machinery/firealarm.dm (hunk 4 above)
Comment (nikothedude) [reply code/game/machinery/firealarm.dm:486]: whoopsie

**Comment by Dorsisdwarf** [comment]
as an ai who likes to turn the lights out to save power thank you so much.

**Comment by nikothedude** [comment]
> as an ai who likes to turn the lights out to save power thank you so much.

didnt add the hotkey, just adding some UX to display it

**Comment by nikothedude** [comment]
all done

File: code/modules/power/apc/apc_main.dm [hunk 7]
```diff
@@ -201,6 +201,18 @@
 	RegisterSignal(SSdcs, COMSIG_GLOB_GREY_TIDE, PROC_REF(grey_tide))
 	update_appearance()
 
+	var/static/list/hovering_mob_typechecks = list(
+		/mob/living/silicon = list(
+			SCREENTIP_CONTEXT_CTRL_LMB = "Toggle power",
+			SCREENTIP_CONTEXT_ALT_LMB = "Toggle equipment power",
+			SCREENTIP_CONTEXT_SHIFT_LMB = "Toggle lighting power",
+			SCREENTIP_CONTEXT_CTRL_SHIFT_LMB = "Toggle environment power"
```
Comment (Mothblocks) [inline code/modules/power/apc/apc_main.dm:209]:
```suggestion
			SCREENTIP_CONTEXT_CTRL_SHIFT_LMB = "Toggle environment power",
```

**Review by Mothblocks — APPROVED** [review]

**Comment by Dorsisdwarf** [comment]
> > as an ai who likes to turn the lights out to save power thank you so much.
> 
> didnt add the hotkey, just adding some UX to display it

yeah but I didnt know it

**Pull request merged**